### PR TITLE
support to specify enabled controllers

### LINKED ISSUE #1379: [feature request]support enable or disable controllers for yurt-manager
**What would you like to be added**:
Like kube-controller-manager component, we can add a parameter `--controllers=-foo,bar,*` for yurt-manager component in order to enable or disable some specified controllers.

at the same time, webhook corresponding to controller should be disabled when controller is disabled.

**Why is this needed**:
End user can disable or enable controllers in yurt-manager according to their requirements.

**others**
/kind feature

### PR DESCRIPTION
**Pull request by xavier-hou**
#### Which issue(s) this PR fixes:

Fixes #1379 

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: charts/openyurt/values.yaml [hunk 1]
```diff
@@ -51,6 +51,8 @@ yurtManager:
     port: 10270
   metrics:
     port: 10271
+  # format should be "foo,-bar"
+  enabledControllers: ""
```
Comment (rambohe-ch) [inline charts/openyurt/values.yaml:55]: enabledControllers --> controllers

File: charts/openyurt/values.yaml [hunk 2]
```diff
@@ -51,6 +51,8 @@ yurtManager:
     port: 10270
   metrics:
     port: 10271
+  # format should be "foo,-bar"
```
Comment (rambohe-ch) [inline charts/openyurt/values.yaml:54]: "foo,-bar" --> "foo,-bar, *"

File: pkg/webhook/server.go [hunk 3]
```diff
@@ -40,33 +41,47 @@ type SetupWebhookWithManager interface {
...
+var controllerWebhook map[string][]SetupWebhookWithManager
 var WebhookHandlerPath = make(map[string]struct{})
 
+func addWebhook(name string, handler SetupWebhookWithManager) {
+	if controllerWebhook == nil {
+		controllerWebhook = make(map[string][]SetupWebhookWithManager)
+	}
+
+	if controllerWebhook[name] == nil {
+		controllerWebhook[name] = make([]SetupWebhookWithManager, 1)
```
Comment (rambohe-ch) [inline pkg/webhook/server.go:53]: make([]SetupWebhookWithManager, 1) --> make([]SetupWebhookWithManager, 0)

**Comment by rambohe-ch** [comment]
@xavier-hou some webhooks maybe have not connected controllers, so we need to add a parameter `--webhooks` for enabling or disabling specified webhooks.

**Comment by xavier-hou** [comment]
> @xavier-hou some webhooks maybe have not connected controllers, so we need to add a parameter `--webhooks` for enabling or disabling specified webhooks.

@rambohe-ch 
1. For contollers who have corresponding webhooks, such as `staticpod`, if the controller is disabled, then the corresponding webhook will be disabled as well. Because we use a map `controllerWebhook` to check whether controller has been disabled.
https://github.com/openyurtio/openyurt/pull/1388/files#diff-c07be401391cf159ca42b3e3c6e8f31d3d0a0a6b1b48998069b64d585a264f35R64 
2. For controllers who don't have corresponding webhooks, then we do nothing.

So I think current PR is enough for enabling and disabling controllers. But If we just want to enable or disable webhooks, then we can add a new parameter `--webhooks`.

File: pkg/webhook/server.go (hunk 3 above)
Comment (xavier-hou) [reply pkg/webhook/server.go]: fixed

File: charts/openyurt/values.yaml (hunk 2 above)
Comment (xavier-hou) [reply charts/openyurt/values.yaml]: done

File: charts/openyurt/values.yaml (hunk 1 above)
Comment (xavier-hou) [reply charts/openyurt/values.yaml]: done

**Comment by xavier-hou** [comment]
@rambohe-ch  We have lots of files aimed at adding controllers to `yurt-manager`, such as `add_csrapprover.go`. They are generated by `newcontroller` command which is convenient for controllers migration but annoying to maintain in the future. How about removing these files and accomplish all controllers register in `controller.go`?

**Comment by rambohe-ch** [comment]
@xavier-hou and i will merge this pr at first.

File: charts/openyurt/templates/yurt-manager.yaml [hunk 4]
```diff
@@ -58,6 +58,9 @@ spec:
             - --logtostderr=true
             - --v={{ .Values.yurtManager.log.level }}
             - --working-namespace={{ include "openyurt.namespace" . }}
+            {{- if .Values.yurtManager.enabledControllers }}
```
Comment (rambohe-ch) [inline charts/openyurt/templates/yurt-manager.yaml:61]: .Values.yurtManager.enabledControllers --> .Values.yurtManager.controllers

File: charts/openyurt/templates/yurt-manager.yaml [hunk 5]
```diff
@@ -58,6 +58,9 @@ spec:
...
+            - --controllers={{ .Values.yurtManager.enabledControllers }}
```
Comment (rambohe-ch) [inline charts/openyurt/templates/yurt-manager.yaml:62]: ditto

File: charts/openyurt/templates/yurt-manager.yaml (hunk 4 above)
Comment (xavier-hou) [reply charts/openyurt/templates/yurt-manager.yaml]: fixed

File: charts/openyurt/templates/yurt-manager.yaml (hunk 5 above)
Comment (xavier-hou) [reply charts/openyurt/templates/yurt-manager.yaml]: fixed

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**